internal/service/clouddiscovery/model_discovery_config.go

```diff
@@ -0,0 +1,274 @@
+			stringplanmodifier.RequiresReplaceIfConfigured(),
+		},
+		Validators: []validator.String{
+			stringvalidator.OneOf("Amazon Web Services", "Google Cloud Platform", "Microsoft Azure"),
+		},
+		MarkdownDescription: "Provider type. Ex.: Amazon Web Services, Google Cloud Platform, Microsoft Azure.",
+	},
+	"source_configs": schema.ListNestedAttribute{
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: SourceConfigResourceSchemaAttributes,
+		},
+		Computed:            true,
+		Optional:            true,
+		MarkdownDescription: "Source configs.",
+	},
+	"status": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "Status of the sync operation. In single account case, Its the combined status of account & all the destinations statuses In auto discover case, Its the status of the account discovery only.",
+	},
+	"status_message": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "Aggregate status message of the sync operation.",
+	},
+	"sync_interval": schema.StringAttribute{
+		Optional:            true,
+		Computed:            true,
+		Default:             stringdefault.StaticString("Auto"),
+		MarkdownDescription: "Sync interval. Default is \"Auto\".",
+	},
+	"tags": schema.MapAttribute{
+		ElementType:         types.StringType,
+		Optional:            true,
+		Computed:            true,
+		Default:             mapdefault.StaticValue(types.MapNull(types.StringType)),
+		MarkdownDescription: "Tagging specifics.",
+	},
+	"tags_all": schema.MapAttribute{
+		ElementType:         types.StringType,
+		Computed:            true,
+		MarkdownDescription: "Tagging specifics includes default tags.",
+	},
+	"updated_at": schema.StringAttribute{
+		CustomType:          timetypes.RFC3339Type{},
+		Computed:            true,
+		MarkdownDescription: "Timestamp when the object has been updated.",
+	},
+}
+
+func (m *DiscoveryConfigModel) Expand(ctx context.Context, diags *diag.Diagnostics, isCreate bool) *clouddiscovery.DiscoveryConfig {
+	if m == nil {
+		return nil
+	}
+	to := &clouddiscovery.DiscoveryConfig{
+		Name:                    flex.ExpandString(m.Name),
+		CredentialPreference:    ExpandCredentialPreference(ctx, m.CredentialPreference, diags),
+		AccountPreference:       flex.ExpandString(m.AccountPreference),
+		ProviderType:            flex.ExpandString(m.ProviderType),
+		AdditionalConfig:        ExpandAdditionalConfig(ctx, m.AdditionalConfig, diags),
+		Description:             flex.ExpandStringPointer(m.Description),
+		DesiredState:            flex.ExpandStringPointer(m.DesiredState),
+		Destinations:            flex.ExpandFrameworkListNestedBlock(ctx, m.Destinations, diags, ExpandDestination),
+		DestinationTypesEnabled: flex.ExpandFrameworkListString(ctx, m.DestinationTypesEnabled, diags),
+		SyncInterval:            flex.ExpandStringPointer(m.SyncInterval),
+		Tags:                    flex.ExpandFrameworkMapString(ctx, m.Tags, diags),
+	}
+	if isCreate {
```

Comment:
isCreate is added as SourceConfig isnt expected in a put call and when passed throws an error for Empty Credential ID.